### PR TITLE
Move log API to stardew.chat

### PR DESCRIPTION
**Pull request by AnotherPillow**
In the event of my server completely dying (such as last night), the log API goes down so I thought it'd be good idea to move it to stardew.chat.

### CODE REVIEW

File: html/log.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    $ini += strlen($start);
+    $len = strpos($string, $end, $ini) - $ini;
+    return substr($string, $ini, $len);
+}
+
+$credits = "Made by AnotherPillow, base PHP code by xintrix (xintrix-dev on github)";
+
+$mc1 = get_string_between($link, '"log-message-text">', '</span>');
+
+$files = glob('../images/*');
```
Comment (aquova) [inline html/log.php:18]: Is this ever used?

File: html/log.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+//$loglink = "https://smapi.io/log/1d07e685d20b4fa2be7907c9b4b3aeb3";
```
Comment (aquova) [inline html/log.php:19]: Remove unused commented out code (here and below)

File: html/log.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+$loglink = $_SERVER["QUERY_STRING"];
+$logid = ltrim($loglink,"https://smapi.io/log/");
+$link = file_get_contents($loglink);
+
+$sdv = get_string_between($link, 'data-game-version="', '"');
+//set success to false if sdv is blank
+if ($sdv == "") {
```
Comment (aquova) [inline html/log.php:26]: We can combine these if/else checks into `$sdvsuccess = $sdv != ""`, both here and elsewhere.

**Review by aquova — CHANGES_REQUESTED** [review]
I left some comments in the code, but I have some questions before we merge this.

1. How intensive do we expect this to be? I don't think this should be too bad, since it's just parsing what smapi.log produces, but since this will be running on my own hardware, I wanted to ensure this isn't going to add some huge overhead. I'm also curious if we need this PHP parsing, or if everything could be done via Python.

2. I discussed with the mods how they felt about this feature as a whole, and the main opinion was that there could be more useful information to display. The SMAPI site displays the number of errors as well as "Suggested fixes", could those be shown instead? I think the consensus is that while the info is somewhat useful, you still have to click on the smapi link anyway, not saving you any time at all.

File: html/log.php (hunk 1 above)
Comment (AnotherPillow) [reply html/log.php:18]: I don't think so, no, I'll remove that.

**Comment by AnotherPillow** [comment]
In response to question 1:
Doing it all in python is a good point, I didn't think of that. It probably wouldn't use a huge amount of resources either.
question 2:
I'll add more things when I have time, was planning to but never got around to it.

**Comment by AnotherPillow** [comment]
Added requested changes

**Comment by AnotherPillow** [comment]
Oops i think i did the line ending thing again
i'll fix that as soon as i can (may be a few days)

**Comment by AnotherPillow** [comment]
fixed

File: src/main.py [hunk 4]
```diff
@@ -181,12 +187,17 @@ async def on_message(message: discord.Message):
...
-        log_dict = requests.get(f"http://api.pil.ninja/smapi_log/endpoint?{log_link}").json()
+        log_dict = requests.get(f"{SERVER_URL}/log.php?{log_link}").json()
         if log_dict["success"]:
             windows_info = re.search(r"(Windows (?:Vista|\d+)) .+", log_dict["OS"])
             if windows_info:  # Condense OS text for Windows because it's often quite verbose.
                 log_dict["OS"] = windows_info.group(1)
-            await message.channel.send(smapi_log_message_template.substitute(log_dict))
+            if log_dict["suggested_fixes"] == []:
+                await message.channel.send(smapi_log_message_template.substitute(log_dict))
+            elif log_dict["suggested_fixes"] != []:
```
Comment (aquova) [inline src/main.py:197]: Just else

File: src/main.py [hunk 5]
```diff
@@ -67,6 +67,12 @@
     "**Log Info:** SMAPI $SMAPI_ver with SDV $StardewVersion on $OS, "
     "with $SMAPIMods C# mods and $ContentPacks content packs."
 )
+smapi_log_message_template_2 = string.Template(
+    "**Log Info:** SMAPI $SMAPI_ver with SDV $StardewVersion on $OS, "
+    "with $SMAPIMods C# mods and $ContentPacks content packs.\n",
```
Comment (aquova) [inline src/main.py:72]: Throws an error with the comma at the end.

File: src/main.py [hunk 6]
```diff
@@ -181,12 +187,17 @@ async def on_message(message: discord.Message):
             await message.channel.send(lvl_up_message)
 
     for log_link in re.findall(r"https://smapi.io/log/[a-zA-Z0-9]{32}", message.content):
-        log_dict = requests.get(f"http://api.pil.ninja/smapi_log/endpoint?{log_link}").json()
+        log_dict = requests.get(f"{SERVER_URL}/log.php?{log_link}").json()
```
Comment (aquova) [inline src/main.py:190]: Tried to run this, and the program throws an error `json.decoder.JSONDecodeError: Expecting value: line 1 column 1`. When I try and open that URL, there is no json file to parse, so something is up with the PHP code.

File: html/log.php [hunk 7]
```diff
@@ -0,0 +1,98 @@
+<?php
```
Comment (aquova) [inline html/log.php:1]: I don't really see why the PHP parsing is necessary at all. I think we should scrap the PHP and just parse the log file all in Python. This just adds an extra layer of complexity.

File: src/main.py (hunk 6 above)
Comment (AnotherPillow) [reply src/main.py:190]: Does the php show an error if you enable errors?

File: html/log.php (hunk 7 above)
Comment (AnotherPillow) [reply html/log.php:1]: Probably a good idea, will do that then (as soon as i can)

File: src/main.py (hunk 4 above)
Comment (AnotherPillow) [reply src/main.py:197]: true

File: src/main.py (hunk 5 above)
Comment (AnotherPillow) [reply src/main.py:72]: will fix that.

**Comment by AnotherPillow** [comment]
Should I just close the pull request and open another with the python stuff rather than making more commits to delete it again?

**Pull request closed without merging**